core/src/test/scala/unit/kafka/message/ByteBufferMessageSetTest.scala

```diff
@@ -422,6 +422,22 @@ class ByteBufferMessageSetTest extends BaseMessageSetTestCases {
     checkWriteFullyToWithMessageSet(createMessageSet(messages))
   }
 
+  @Test
+  def testIterationOverCompressedSet(): Unit = {
+    val messageCount = 100
+    val messages = (1 to messageCount).map(m => {
+      new Message(s"message num $m".getBytes)
+    })
+    var messageSet = new ByteBufferMessageSet(GZIPCompressionCodec, messages:_*)
```

Comment:
after actually doing this i realized that there are 13 other test methods in this class, that would runx4 times for no reason if i parameterize the class just for my test. you still want to do this?